docs/internals/optimizer.rst

```diff
@@ -832,10 +852,13 @@ value, one of them will always be unused. The UnusedPruner or the
 UnusedAssignEliminator will then be able to fully eliminate such
 variables.
 
-This step is especially efficient if the ExpressionSplitter is run
-before. If the code is in pseudo-SSA form,
-the values of variables are available for a longer time and thus we
-have a higher chance of expressions to be replaceable.
+The longer the values of variables are available, the higher the chance of an expression being replaceable.
+For this reason the step could potentially be more effective when used on code in pseudo-SSA form.
+This property, however, is not easily destroyed once established, and in most cases it is enough
+if SSATransform was used once at any point before running the step.
+The difference may also not be very significant in practice.
```

Comment:
The difference of what vs what? If it is running SSA immediately beforehand vs just somewhere before, I suggest to drop this sentence